.gitlab-ci.yml

```diff
@@ -0,0 +1,38 @@
+variables:
+  CI_IMAGE: registry.ddbuild.io/images/docker:20.10.13
+  RESTORE_CACHE_ATTEMPTS: 2
+
+stages:
+  - build
+
+# =======================================================================
+# Build and deploy the images used for CI
+# =======================================================================
+
+.build-ci-image: &build-ci-image
+  stage: build
+  tags: ["runner:docker"]
+  image: $CI_IMAGE
+  script:
+    - TAG="$IMAGE_TAG_PREFIX-v$CI_COMMIT_SHORT_SHA"
+    - docker build --file $DOCKERFILE --tag registry.ddbuild.io/ci/opentelemetry-collector-contrib:$TAG --label target=staging .
+    - docker push registry.ddbuild.io/ci/opentelemetry-collector-contrib:$TAG
```

Comment:
You can remove the script, and add the variables, yielding a common anchor for both:
```
build-ci-image-main:
    <<: *build-ci-image
    script:
        ...

build-ci-image-demo:
    <<: *build-ci-image
    script:
        ...
```